apps/ui/components/reservation-unit/Head.tsx

```diff
@@ -107,6 +107,38 @@ const UnitName = styled(H3).attrs({ as: "h2" })`
   }
 `;
 
+const NonReservableNotification = ({
+  reservationUnit,
+}: {
+  reservationUnit: ReservationUnitByPkType;
+}) => {
+  const { t } = useTranslation();
+  const returnText = t(
+    `reservationUnit:notifications.${
+      reservationUnit.reservationKind ===
+      ReservationUnitsReservationUnitReservationKindChoices.Season
+        ? "onlyRecurring"
+        : isReservationStartInFuture(reservationUnit)
+          ? t("reservationUnit:notifications.futureOpening", {
+              date: reservationUnit.reservationBegins
+                ? formatDate(reservationUnit.reservationBegins, "d.M.yyyy")
+                : "",
+              time: reservationUnit.reservationBegins
+                ? formatDate(reservationUnit.reservationBegins, "H.mm")
+                : "",
+            })
+          : "notReservable"
+    }`
+  );
```

Comment:
This is too hard to follow (nested ternaries). I'd separate the error message generation to a function, but anything other than multiple ternaries (define the messages as variables first, then pick the right one in one select etc.).